laravel/app/AutoAssignment.php

```diff
@@ -11,287 +11,212 @@
-
-        foreach ($remainder as $other) {
-            $preference = Option::where('date', $date)->where('resident', $other)->where('option', $option)->where('isValid',"1")->value('schedule');
-		Log::info($preference.' preference');
-            if (is_null($preference)) {
-                array_push($newRemainder, $other);
-                continue;
-            }
-
-            if (Option::where('date', $date)->where('schedule', $preference)->where('option', $option)->where('isValid',"1")->count() == 1) {
-                //add unique preferences to database
-		Log::info("unique pref");
-                self::addAssignment($preference, $other, $date);
-                self::updateProbability($other, $tickets);
+            // Residents on same rotation as schedule get priority
+            // Case 1: if only 1 resident on same rotation
+            if(count($onRotation) == 1){
+                // residents not on rotation are added to remainder array
+                $remainder = array_merge($remainder, $notOnRotation);
+                // assign resident to schedule
+                self::addAssignment($wantedSchedule, $onRotation[0]['resident'], $date);
+                self::updateProbability($onRotation[0]['resident'], $ticketsToAdd);
             }
-            else if (!in_array($preference, $candidates)) {
-                array_push($candidates, $preference);
-            }
-        }
-
-        //determine who gets there preference when multiple people want the same schedule
-        foreach ($candidates as $candidate) {
-            $schedRotation = ScheduleData::where('id', $candidate)->value('rotation');
-	    Log::info($schedRotation);
-            $competitors = Option::where('date', $date)->where('schedule', $candidate)->where('option', $option)->where('isValid',"1")->get();
-            $maxScore=-1;
-            $toPush = null;
-            $rotationMatches = array();
-            $notOnRotation = array();
-
-            //get each resident's rotation
-            foreach ($competitors as $competitor) {
-              $resname = Resident::where('id', $competitor['resident'])->value('name');
-              $resRotations = Rotations::where('name', $resname)->get();
-              foreach ($resRotations as $resRotation) {
-                if (($date >= $resRotation['Start']) && ($date <= $resRotation['End'])) {
-                    $serviceId = $resRotation['Service'];
+            // Case 2: multiple residents on same rotation 
+            else if(count($onRotation) > 1){
+                // find which resident on rotation has the most tickets
+                foreach($onRotation as $onRotationOption){
+                    // add losing residents to remainder array
+                    if($maxTickets < Probability::where('resident', $onRotationOption['resident'])->value('total')){
+                        $maxTickets = Probability::where('resident', $onRotationOption['resident'])->value('total');
```

Comment:
This will query the database 2x for the same value. Change to:

```php
$tickets = Probability::where('resident', $onRotationOption['resident'])->value('total');
if($tickets > $maxTickets) {
    $maxTickets = $tickets;
    // other statements
}
```